softgroup/services/profile/profile-impl/src/main/java/com/softgroup/profile/impl/handler/GetOtherProfilesHandler.java

```diff
@@ -17,12 +25,47 @@
 public class GetOtherProfilesHandler
         extends AbstractRequestHandler<GetOtherProfilesRequest,
             GetOtherProfilesResponse> implements ProfileRequestHandler {
+
+    @Autowired
+    private ProfileService profileService;
+
+    @Autowired
+    private Mapper mapper;
+
     public String getName() {
         return "get_other_profiles";
     }
 
     @Override
     public Response<GetOtherProfilesResponse> doHandle(Request<GetOtherProfilesRequest> request) {
-        return null;
+        GetOtherProfilesRequest requestData = request.getData();
+        GetOtherProfilesResponse getOtherProfilesResponse = new GetOtherProfilesResponse();
+
+
+        List<String> uuids = requestData.getUserID();
+        List<ProfileDTO> profileDTOS = getProfileDTOs(uuids);
+
+        getOtherProfilesResponse.setProfileDTOS(profileDTOS);
+
+        Response<GetOtherProfilesResponse> response = new Response<GetOtherProfilesResponse>();
+        response.setHeader(request.getHeader());
+        response.setData(getOtherProfilesResponse);
+
+        ResponseStatus status = new ResponseStatus();
+        status.setCode(200);
+        status.setMessage("OK");
+
+        response.setStatus(status);
+        return response;
+    }
+
+    private List<ProfileDTO> getProfileDTOs(List<String> uuids) {
+        List<ProfileDTO> profileDTOS = new ArrayList<>();
+
+        for (String uuid : uuids) {
+            profileDTOS.add(mapper.mapProfile(profileService.findProfileById(uuid)));
```

Comment:
@VadymPolyanski Use find by array of ids. That will fetch data by one request. Here you will make as many SQL requests as you have ids in request